src/components/generator/GenerateDialog.js

```diff
@@ -0,0 +1,98 @@
+import Notifications from "../common/Notifications";
+import {Button, Dialog, DialogContent, DialogTitle, IconButton, styled} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import Settings from "../common/settings";
+import { useEffect, useState } from "react";
+import FetchHandler from "../common/FetchHandler";
+
+const GenerateDialogRoot = styled('div')({
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+
+  '> *': {
+    marginBottom: '16px',
+  }
+});
+
+const StyledButton = styled(Button)({
+    cursor: 'pointer',
+    width: '128px',
+    display: 'flex',
+
+    ':hover': {
+        backgroundColor: '#BFBFBF',
+    },
+});
+
+const StyledA = styled('a')({
+  textDecoration: "none",
+});
+
+const StyledIconButton = styled(IconButton)({
+    position: 'absolute',
+    top: 10,
+    right: 0,
+});
+
+const StyledDialogContent = styled(DialogContent)({
+    display: "flex",
+    justifyContent: "center"
+});
+
+const EditDialog = ({setGenerate, open, setOpen, selectedProject, setSelectedProject}) => {
+    const [projectRetrievalState, setProjectRetrievalState] = useState("loading");
+
+    useEffect(() => {
+        FetchHandler.simple(fetch(Settings.generatePath + selectedProject.id, {method: "GET"}))
+            .then(() => {
+                setProjectRetrievalState("success");
+            })
+            .catch((err) => {
+                Notifications.notify(`Failed to retrieve data\n${err}`, "error");
+                setProjectRetrievalState("failed");
+            });
+      }, [selectedProject]);
+
+    const closeGenerate = () => {
+        setSelectedProject(undefined);
+        setOpen(false);
+        setGenerate(false);
+    };
+
+    return (
+        <GenerateDialogRoot>
+            <Dialog
+                open={open}
+                onClose={closeGenerate}
+                maxWidth={'xs'}
```

Comment:
```suggestion
                maxWidth="xs"
```
Static property suffices

Reply:
implemented in branch: "generatorpage-add-delete"